instrumentation/aws_lambda/lib/opentelemetry/instrumentation/aws_lambda/handler.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module AwsLambda
+      # Handler class that creates a span around the _HANDLER
+      class Handler
+        attr_reader :handler_method, :handler_class
+
+        # anytime when update the code in a Lambda function or change the functional configuration,
+        # the next invocation results in a cold start; therefore these instance variable will be up-to-date
+        def initialize
+          @flush_timeout    = ENV.fetch('OTEL_INSTRUMENTATION_AWS_LAMBDA_FLUSH_TIMEOUT', '30000').to_i
+          @original_handler = ENV['ORIG_HANDLER'] || ENV['_HANDLER'] || ''
+          @handler_class    = nil
+          @handler_method   = nil
+          @handler_file     = nil
+
+          resolve_original_handler
+        end
+
+        # We want to capture the error if user's handler is causing issue
+        # but our wrapper and handler shouldn't cause any issue
+        def call_wrapped(event:, context:)
+          parent_context   = extract_parent_context(event)
+          span_attributes  = otel_attributes(event, context)
+
+          response = nil
+          original_handler_error = nil
+          begin
+            response = call_original_handler(event: event, context: context)
+            status_code = response.is_a?(Hash) ? (response['statusCode'] || response[:statusCode]) : nil
+          rescue StandardError => e
+            original_handler_error = e
+            status_code = '500'
```

Comment:
I don't think instrumentation can assume this is an HTTP invocation trigger since there are [many trigger types](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/faas/faas-spans.md#function-trigger-type) and blindly collect a status code attribute.

Reply:
Updated the part that checking the type of trigger. Thanks!